lib/std/fs.zig

```diff
@@ -928,6 +928,24 @@ pub const Dir = struct {
         return self.openDir(sub_path, open_dir_options);
     }
 
+    /// This function returns the canonicalized absolute pathname of `pathname`
+    /// relative to this `Dir`. If `pathname` is absolute, ignores this `Dir` handle
+    /// and returns the canonicalized absolute pathname of `pathname` argument.
+    /// Caller must free the returned memory.
+    pub fn realpath(self: Dir, allocator: *Allocator, pathname: []const u8) ![]u8 {
```

Comment:
Following conventions, I think this should be `realpathAlloc` and there should be a `realpath`, `realpathZ`, and `realpathW` that take a slice as a buffer. See `std.fs.realpath` and friends:

https://github.com/ziglang/zig/blob/bb55889ce0989e120b85a38aec44865a20844ff0/lib/std/fs.zig#L26-L29

Reply:
Excellent point, thanks! Will tweak and report back.

Reply:
@squeek502 I've now added first pass at implementing your suggestion. It'd be great if you could have a look and lemme know if there's any missing, or if something could be improved upon, etc.

Reply:
I didn't realize that the `os.realpath` functions use `*[MAX_PATH_BYTES]u8` for their out_buffer params. Changing `Dir.realpath` to use that instead of `[]u8` would allow for removing the intermediate buffers in their implementations which seems useful. Probably worth waiting for @andrewrk to chime in on that before changing it though.

Reply:
Oh, yeah, that’s correct. It’d be great to change the interface from array to slice in `std.os.realpath` and friends but my suggestion would be doing it in a subsequent PR. Would that make sense?